File: examples/vec2.go
package main

import "fmt"

type Vec2 struct {
	X int
	Y int
}

func (v Vec2) plus(other Vec2) Vec2 {
	return Vec2{v.X + other.X, v.Y + other.Y}
}

func (v *Vec2) moveX(delta int) {
	v.X = v.X + delta
}

func main() {
	a := Vec2{1, 2}
	b := Vec2{3, 4}
	c := a.plus(b)
	c.moveX(10)
	fmt.Println(c.X)
	fmt.Println(c.Y)
}

File: translated/vec2.java
public record Vec2(int x, int y) {
    public Vec2 plus(Vec2 other) {
        return new Vec2((x() + other.x), (y() + other.y));
    }
    public static void main(String[] args) {
        var a = new Vec2(1, 2);
        var b = new Vec2(3, 4);
        var c = a.plus(b);
        System.out.println(c.x);
        System.out.println(c.y);
    }
}
